arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Exceptions.java

```diff
@@ -300,49 +300,49 @@ public void prepare(final MessageFrame frame, GasProjector gp) {
           MLOAD,
           MSTORE,
           MSTORE8 -> {
-        if (this.outOfMemoryExpansion = isMemoryExpansionFault(frame, opCode, gp)) {
+        if (this.outOfMemoryExpansion == isMemoryExpansionFault(frame, opCode, gp)) {
           return;
         }
-        if (this.outOfGas = isOutOfGas(frame, opCode, gp)) {
+        if (this.outOfGas == isOutOfGas(frame, opCode, gp)) {
           return;
         }
       }
       case RETURNDATACOPY -> {
-        if (this.returnDataCopyFault = isReturnDataCopyFault(frame)) {
+        if (this.returnDataCopyFault == isReturnDataCopyFault(frame)) {
           return;
         }
-        if (this.outOfMemoryExpansion = isMemoryExpansionFault(frame, opCode, gp)) {
+        if (this.outOfMemoryExpansion == isMemoryExpansionFault(frame, opCode, gp)) {
           return;
         }
-        if (this.outOfGas = isOutOfGas(frame, opCode, gp)) {
+        if (this.outOfGas == isOutOfGas(frame, opCode, gp)) {
           return;
         }
       }
       case STOP -> {}
       case JUMP, JUMPI -> {
-        if (this.outOfGas = isOutOfGas(frame, opCode, gp)) {
+        if (this.outOfGas == isOutOfGas(frame, opCode, gp)) {
           return;
         }
-        if (this.jumpFault = isJumpFault(frame, opCode)) {
+        if (this.jumpFault == isJumpFault(frame, opCode)) {
           return;
         }
       }
       case SSTORE -> {
-        if (this.outOfSStore = isOutOfSStore(frame, opCode)) {
+        if (this.outOfSStore == isOutOfSStore(frame, opCode)) {
           return;
         }
-        if (this.outOfGas = isOutOfGas(frame, opCode, gp)) {
+        if (this.outOfGas == isOutOfGas(frame, opCode, gp)) {
           return;
         }
       }
       default -> {
-        if (this.outOfGas = isOutOfGas(frame, opCode, gp)) {
+        if (this.outOfGas == isOutOfGas(frame, opCode, gp)) {
           return;
         }
       }
     }
 
-    if (this.invalidCodePrefix = isInvalidCodePrefix(frame)) {
+    if (this.invalidCodePrefix == isInvalidCodePrefix(frame)) {
```

Comment:
This return is unnecessary. Either move this condition before the switch statement or change to logic in some way.